MatrixSDK/Threads/MXThreadModel.swift

```diff
@@ -0,0 +1,94 @@
+// 
+// Copyright 2021 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+@objcMembers
+public class MXThreadModel: NSObject, MXThreadProtocol {
+
+    public let id: String
+
+    public let roomId: String
+
+    public let notificationCount: UInt
+
+    public let highlightCount: UInt
+
+    public let isParticipated: Bool
+
+    public let rootMessage: MXEvent?
+
+    public private(set) var lastMessage: MXEvent?
+
+    public private(set) var numberOfReplies: Int
+
+    public init(withRootEvent rootEvent: MXEvent,
+                notificationCount: UInt = 0,
+                highlightCount: UInt = 0) {
+        self.id = rootEvent.eventId
+        self.roomId = rootEvent.roomId
+        self.notificationCount = notificationCount
+        self.highlightCount = highlightCount
+        self.rootMessage = rootEvent
+        if let thread = rootEvent.unsignedData?.relations?.thread {
```

Comment:
Should this maybe be a throwing constructor for when the passed root event is invalid?

Reply:
That would make perfect sense, but i don't want to do it at this point, due to issue for redacted events: https://github.com/matrix-org/synapse/issues/11753#issuecomment-1032693709

Reply:
Fair enough 👍